content/articles/eclat-algorithm-in-python/index.md

```diff
@@ -0,0 +1,154 @@
+author: jackson-munyai
+date: 2021-12-17T00:00:00-21:00
+topics: [Machine Learning]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/eclat-algorithm-in-python/hero.png 
+    alt: Getting Started with ECLAT Algorithm Example Image
+---
+The goal in data mining is to find rules that predict an item based on the occurrence of other items in the transactions.
+<!--more-->
+
+Generally, to achieve this, we have two approaches;
+1. Generating the frequent itemsets
+Here we generate all itemsets with greater support than the adopted minimum support.
+2. Rule Generation
+Generate a frequent high set from the frequent itemsets.
+
+### Prerequisites:
+To follow up with this article, the learner needs to be comfortable working with the following libraries in python, Sk-learn, Pandas, Numpy.
+
+### Introduction to ECLAT Algorithm
+As we foresaid, the main idea in the association rule is to discover valid information and knowledge from a large dataset. There number of algorithms that have been developed over the years that make this activity as successful as possible. The major algorithm used includes;
+
+1. Apriori Algorithm
+2. Eclat Algorithm
+3. FP Growth Algorithm
+
+The first algorithm to be introduced in the Data Mining domain was the Apriori algorithm. This algorithm, however, had some limitations in discovering frequent itemsets. Its limitations created a need for a more efficient algorithm.
+
+Later, the Eclat algorithm was introduced to deal with the weakness of the Apriori algorithm. Between these two algorithms, we have a significant difference.
+
+The first difference is that, unlike the Apriori algorithm, which is applicable with *horizontal dataset*, the Eclat algorithm is applicable only with a dataset in *verticle dataset* format.
+
+Secondly, in the Eclat algorithm, only the *support* is counted, and the confidence, as in the case of Apriori, is not computed. Here, the *Support* is nothing but the number of times an item is in a database.
+
+At each stage of the generated database, the Eclat algorithm uses the current generated dataset to learn frequent itemset, unlike the Apriori, which scans the original database repeatedly. Since the Eclat scan over the database once, it is much faster than the Apriori algorithm.
+
+However, this doesn't mean the Apriori algorithm is worse. On the contrary, when dealing with a larger dataset, Apriori tends to shine best. Therefore, the Eclat algorithm works better with small and medium datasets.
+
+The key takeaway here is that Eclat works with a vertical data format. Since most datasets are in horizontal format, to apply the Eclat algorithm, we first have to convert them to vertical format.
+
+Below are examples of Horizontal and verticle data formats;
+
+![Horizontal data](/engineering-education/eclat-algorithm-in-python/horizontal.png)
+Horizontal data format
+
+![Verticle data](/engineering-education/eclat-algorithm-in-python/verticle-format.png)
+Verticle data format
```

Comment:
Please confirm to me that the image names correspond to their respective images.

Reply:
They are similar.